protocols/identify/src/handler.rs

```diff
@@ -318,11 +343,14 @@ impl ConnectionHandler for Handler {
         {
             self.inbound_identify_push.take();
 
-            if let Ok(info) = res {
-                self.update_supported_protocols_for_remote(&info);
-                return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(Event::Identified(
-                    info,
-                )));
+            if let Ok(remote_push_info) = res {
+                if let Some(info) = self.patch_incoming_push_info(remote_push_info) {
+                    self.handle_incoming_info(&info);
+
+                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
+                        Event::Identified(info),
+                    ));
+                };
```

Comment:
Is it worth having all of `patch_incoming_push_info` if we could also do:

```suggestion
                if let Some(mut info) = self.remote_info.clone() {
                	let info = info.merge(remote_push_info);
                    self.handle_incoming_info(&info);

                    return Poll::Ready(ConnectionHandlerEvent::NotifyBehaviour(
                        Event::Identified(info),
                    ));
                };
```

I don't think the log adds much value. It is not like the user can do anything against that and I am also pretty sure it is super unlikely to hit because we request identify immediately on connection setup.